pkg/resources/management.cattle.io/v3/project/validator.go

```diff
@@ -249,3 +302,15 @@ func formatResourceList(resources v1.ResourceList) string {
 	sort.Strings(resourceStrings)
 	return strings.Join(resourceStrings, ",")
 }
+
+func parseResource(s string) (resource.Quantity, error) {
+	if s == "" {
+		// Not a literal "0".
+		return resource.Quantity{}, nil
+	}
+	return resource.ParseQuantity(s)
+}
+
+func resourceSpecified(q resource.Quantity) bool {
```

Comment:
I dislike this function since it assumes that `resource.ParseQuantity()` won't return a `resource.Quantity{}`. Given that we don't do any direct comparison of the values here (i.e. we are using `resource.Quantity.Cmp()`) I don't think that this is a safe call to make. Can you elaborate on why you don't return a `*resource.Quantity` and use a nil check instead?

Reply:
It does assume that `resource.ParseQuantity` will return a `resource.Quantity`.
My `parseResource` and `resourceSpecified` functions are meant to be used together.

`parseResource` is a wrapper around `resource.ParseQuantity()`. On its own, `resource.ParseQuantity()` returns an error when it gets an empty string. Thus we need that wrapper. And `resourceSpecified` simply compares a given quantity with its zero value, which is what I chose to use as a sign of a zero-value, unspecified quantity. What is unsafe here?

The reason for all this is the need to know if a value is invalid or just hasn't been specified.

Reply:
> It does assume that `resource.ParseQuantity` will return a `resource.Quantity`

To be clear, I'm not talking about the type here. I'm referring to the fact that you are relying on the parse function not to return the default value, since you assume in your function that the default value means that the quantity was not specified. It's that specific behavior that I think is dangerous.